src/targets/maven.ts

```diff
@@ -444,4 +391,105 @@ export class MavenTarget extends BaseTarget {
+  // Based on: https://github.com/vanniktech/gradle-maven-publish-plugin/ implementation.
+  public async closeAndReleaseRepository(): Promise<void> {
+    const repository = await this.getRepository();
+    const { repositoryId, type } = repository;
+
+    if (type !== 'open') {
+      throw new Error(
+        'No open repositories available. Go to Nexus Repository Manager to see what happened.'
+      );
+    }
+
+    await this.closeRepository(repositoryId);
+    await this.releaseRepository(repositoryId);
+  }
+
+  private getNexusRequestHeaders(): Record<string, string> {
+    return {
+      Accept: 'application/json',
+      Authorization: `Basic ${Buffer.from(
+        `${process.env.OSSRH_USERNAME}:${process.env.OSSRH_USERNAME}`
+      ).toString(`base64`)}`,
+    };
+  }
+
+  private async getRepository(): Promise<NexusRepository> {
+    const response = await fetch(`${NEXUS_API_BASE_URL}/profile_repositories`, {
+      headers: this.getNexusRequestHeaders(),
+    });
+    if (!response.ok) {
+      throw new Error(
+        `Unable to fetch repository: ${response.status}, ${response.statusText}`
+      );
+    }
+
+    const body = await response.json();
+    const repositories = body.data;
+
+    if (repositories.length === 0) {
+      throw new Error(`No available repositories. Nothing to publish.`);
+    }
+
+    if (repositories.length > 1) {
+      throw new Error(
+        `There are more than 1 active repositories. Please close unwanted deployments.`
+      );
+    }
+
+    return repositories[0];
+  }
+
+  private async closeRepository(repositoryId: string): Promise<void> {
+    const response = await fetch(`${NEXUS_API_BASE_URL}/bulk/close`, {
+      headers: this.getNexusRequestHeaders(),
+      method: 'POST',
+      body: JSON.stringify({
+        data: { stagedRepositoryIds: [repositoryId] },
+      }),
+    });
+
+    if (!response.ok) {
+      throw new Error(
+        `Unable to close repository: ${response.status}, ${response.statusText}`
+      );
+    }
+
+    const poolingStartTime = Date.now();
+
+    while (true) {
+      if (Date.now() - poolingStartTime > NEXUS_RETRY_DEADLINE) {
+        throw new Error('Deadline for Nexus repository status change reached.');
+      }
+
+      await sleep(NEXUS_RETRY_DELAY);
+
+      const repository = await this.getRepository();
+      const { type, transitioning } = repository;
+
+      if (type === 'closed' && !transitioning) {
+        break;
+      }
+    }
+  }
+
+  private async releaseRepository(repositoryId: string): Promise<void> {
+    const response = await fetch(`${NEXUS_API_BASE_URL}/bulk/promote`, {
+      headers: this.getNexusRequestHeaders(),
+      method: 'POST',
+      body: JSON.stringify({
+        data: { stagedRepositoryIds: [repositoryId] },
+      }),
+    });
+
+    if (!response.ok) {
+      throw new Error(
+        `Unable to release repository: ${response.status}, ${response.statusText}`
```

Comment:
Should we include the repository ID? Does it help when manual action is required? Is there any security reason to not expose it?

Reply:
There can be only a single repository during this process, so it's not possible when observed instantly, however it can be useful for reference when you want to do something after a weekend or something. Added.